dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageCheckReport.java

```diff
@@ -48,4 +60,98 @@ public String getErrorString() {
+          } else if (symbol instanceof MethodSymbol) {
+            MethodSymbolReference methodSymbolReference =
+                MethodSymbolReference.fromSymbol(classFile, (MethodSymbol) symbol);
+            SymbolNotResolvable<MethodSymbolReference> symbolNotResolvable =
+                createSymbolResolvable(
+                    methodSymbolReference, symbolProblem, targetClassLocation, isReachable);
+            methodSymbolProblemsBuilder.put(jar, symbolNotResolvable);
+          } else if (symbol instanceof FieldSymbol) {
+            FieldSymbolReference fieldSymbolReference =
+                FieldSymbolReference.fromSymbol(classFile, (FieldSymbol) symbol);
+            SymbolNotResolvable<FieldSymbolReference> symbolNotResolvable =
+                createSymbolResolvable(
+                    fieldSymbolReference, symbolProblem, targetClassLocation, isReachable);
+            fieldSymbolProblemsBuilder.put(jar, symbolNotResolvable);
+          }
+        });
+
+    ImmutableSetMultimap<Path, SymbolNotResolvable<ClassSymbolReference>> jarToClassSymbolProblems =
+        classSymbolProblemsBuilder.build();
+    ImmutableSetMultimap<Path, SymbolNotResolvable<MethodSymbolReference>>
+        jarToMethodSymbolProblems = methodSymbolProblemsBuilder.build();
+    ImmutableSetMultimap<Path, SymbolNotResolvable<FieldSymbolReference>> jarToFieldSymbolProblems =
+        fieldSymbolProblemsBuilder.build();
+
+    for (Path jar : jars) {
+      linkageReportBuilder.add(
+          JarLinkageReport.builder()
+              .setJarPath(jar)
+              .setMissingClassErrors(jarToClassSymbolProblems.get(jar))
+              .setMissingMethodErrors(jarToMethodSymbolProblems.get(jar))
+              .setMissingFieldErrors(jarToFieldSymbolProblems.get(jar))
+              .build());
+    }
+
+    return create(linkageReportBuilder.build());
+  }
+
+  private static <U extends SymbolReference> SymbolNotResolvable<U> createSymbolResolvable(
+      U symbolReference,
+      SymbolProblem symbolProblem,
+      Path targetClassLocation,
+      boolean isReachable) {
+    switch (symbolProblem.getErrorType()) {
+      case INACCESSIBLE_CLASS:
+        return SymbolNotResolvable.errorInaccessibleClass(
+            symbolReference, targetClassLocation, isReachable);
+      case INCOMPATIBLE_CLASS_CHANGE:
+        return SymbolNotResolvable.errorIncompatibleClassChange(
+            symbolReference, targetClassLocation, isReachable);
+      case INACCESSIBLE_MEMBER:
+        return SymbolNotResolvable.errorInaccessibleMember(
+            symbolReference, targetClassLocation, isReachable);
+      case SYMBOL_NOT_FOUND:
+        return SymbolNotResolvable.errorMissingMember(
+            symbolReference, targetClassLocation, isReachable);
+      case CLASS_NOT_FOUND:
+        return SymbolNotResolvable.errorMissingTargetClass(symbolReference, isReachable);
+      default:
+        throw new UnsupportedOperationException(
```

Comment:
I'm not sure this is the right exception here, though I can't put my finger on one that fits. What if you moved this method into the SymbolNotResolvable type instead?

Reply:
Moved the function to SymbolNotResolvable and updated the exception to VerifyException (as per [ConditionalFailuresExplained](https://github.com/google/guava/wiki/ConditionalFailuresExplained#summary); "Someone I depend on messed up.")